spyder/app/mainwindow.py

```diff
@@ -2253,6 +2252,7 @@ def main(options, args):
                                       CONF.get('main', 'high_dpi_scaling'))
 
     # **** Set debugging info ****
+    delete_debug_log_files()

```

Comment:
Why did you move this call here? Also, you didn't add the logger call that was present before. Please restore that.

Reply:
The spyder-debug log file from a previous session must be deleted _before_ `setup_logging` is called, otherwise it will be deleted after the logger opens the file resulting in no log file.